spec/lib/decision_date_and_cost_migration_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+RSpec.describe DecisionDateAndCostMigration do
+  subject(:migration) { described_class.new }
+
+  let!(:application1) do
+    create(:application_full_remission, :processed_state, fee: 500, decision: 'full', decision_date: nil, decision_cost: nil)
+  end
+
+  let!(:application2) do
+    create(:application_part_remission, :processed_state, decision_type: 'part_payment', fee: 500, amount_to_pay: 100, decision: 'part', decision_date: nil, decision_cost: nil) do |a|
+      create(:part_payment_part_outcome, :completed, application: a)
+    end
+  end
+
+  let!(:application3) do
+    create(:application_part_remission, :processed_state, decision_type: 'part_payment', fee: 500, amount_to_pay: 100, decision: 'part', decision_date: nil, decision_cost: nil) do |a|
+      create(:evidence_check_part_outcome, :completed, amount_to_pay: 150, application: a)
+      create(:part_payment_part_outcome, :completed, application: a)
+    end
+  end
+
+  let!(:application4) do
+    create(:application_full_remission, :processed_state, decision_type: 'evidence_check', fee: 500, decision: 'full', decision_date: nil, decision_cost: nil) do |a|
+      create(:evidence_check_full_outcome, :completed, application: a)
+    end
+  end
+
+  let!(:application5) do
+    create(:application_no_remission, :processed_state, fee: 500, decision: 'none', decision_date: nil, decision_cost: nil)
+  end
+
+  describe '#run!' do
+    before do
+      migration.run!
+    end
+
+    it 'sets the correct decision_date and decision_cost' do
+      # for processed application without evidence check or part payment
```

Comment:
So the reason why it's in a single test is that I need to make sure that the migration runs on the whole DB at the same time, but each individual scenario is valid. If I tested them in isolation, it wouldn't be guaranteed.

I totally agree though that it's not ideal. On the other hand all the 3 migration classes will be deleted once run and verified.